### ISSUE TITLE
R: Use Cairo device on all systems

### ISSUE THREAD

**Issue by thomasp85** [posit-dev/positron#2919]
While we wait for a potential future where positron uses ragg for plot rendering, we should at least make sure that the png device uses Cairo on all systems, instead of relying on the default.

On linux Cairo is already the default so this would not change anything

On windows the default is a monstrosity and should be avoided at all costs. Using Cairo there is an act of mercy

On Mac, the default quartz device have generally been pretty good but now it consistently crashes the R process when using some of the newer features. Paul Murrell has been made aware, but it would be better to avoid the possibility of users hitting this.

Apart from fixing issues on Mac and Windows there is also a certain neatness in having comparably equivalent output on all OS's

**Comment by juliasilge** [comment]
Request for using ragg in https://github.com/posit-dev/positron/discussions/3803

**Comment by cpsievert** [comment]
Just to make sure I understand, is the current behavior essentially just a call to `grDevices::png()`?

**Comment by lionel-** [comment]
yep see [renderWithPlotDevice()](https://github.com/posit-dev/ark/blob/4ea2c076fcefb2b3012f04df946bf992c29dc17d/crates/ark/src/modules/positron/graphics.R#L29)

**Comment by DavisVaughan** [comment]
Here's where we stand:

**For png, jpeg, and tiff, if you have ragg installed then it will be used for any OS, and you don't have to worry about this at all. This is a large amount of our userbase.**

If you _don't_ have ragg installed:
- On Linux, we already default to `"cairo"`
- On Windows, we already default to `"cairo"`
- On Mac, we currently default to `"quartz"`

We considered changing Mac to `"cairo"` as well, but that gets us back to this problem https://github.com/posit-dev/positron/issues/913. Confusingly, on Mac, using `"cairo"` also requires xquartz as a dependency, which you get with `brew install --cask xquartz` or from downloading it from R's xquartz installer, neither of which make for a good story for R users.

Even more confusingly, the `"quartz"` device on mac doesn't need xquartz, here is some context from Thomas on this:

> Quartz is the legacy mac UI framework, xquartz is bindings between the X11 Linux UI framework and quartz so that linux apps can be cross-compiled on mac. So, the quartz device doesn't need xquartz because quartz is there already, but cairo needs it to interact with the OS

This is why `"quartz"` works out of the box with a fresh Mac and fresh install of R, and is why we have it as the default currently.

Even _more_ confusingly, `capabilities("cairo")` will return `TRUE` on a Mac even if you don't have xquartz installed, so that is not a reliably indicator of whether or not your Mac currently supports `"cairo"`.

To summarize:
- The current default fallback of `"quartz"` on Mac is not that bad. Most issues with R's newer graphics features have been resolved by now, and users will be using ragg as the default if it is installed, so they are less likely to have much trouble with this anyways.
- Switching to a default fallback of `"cairo"` on Mac seems error prone, and hard to get right, and is unlikely to be useful for many people because it requires an external xquartz dependency.

Because of these two bullets, we've decided to stick with `"quartz"` as the default fallback on Mac, and I've used https://github.com/posit-dev/ark/pull/800 as a place to document all of this knowledge in the ark code base.
